codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java

```diff
@@ -74,15 +74,6 @@ public void addConfigInterfaceFields(
             SymbolProvider symbolProvider,
             TypeScriptWriter writer
     ) {
-        writer.addImport("Provider", "__Provider", TypeScriptDependency.AWS_SDK_TYPES.packageName);
-        writer.addImport("Logger", "__Logger", TypeScriptDependency.AWS_SDK_TYPES.packageName);
-
-        writer.writeDocs("Value for how many times a request will be made at most in case of retry.")
-                .write("maxAttempts?: number | __Provider<number>;\n");
-        writer.writeDocs("Specifies which retry algorithm to use.")
-                .write("retryMode?: string | __Provider<string>;\n");
-        writer.writeDocs("Optional logger for logging debug/info/warn/error.")
-                .write("logger?: __Logger;\n");
         writer.writeDocs("Enables IPv6/IPv4 dualstack endpoint.")
                 .write("useDualstackEndpoint?: boolean | __Provider<boolean>;\n");
         writer.writeDocs("Enables FIPS compatible endpoints.")
```

Comment:
While we are updating this, let's make the Integration be appropriately conditional. On the aws.api#service trait.

Reply:
In fact, given these are AWS specific, maybe they should just be folded into AddAwsRuntimeConfig.

Reply:
Moved into AddAwsRuntimeConfig and made appropriately conditional in [3e1cb31](https://github.com/aws/aws-sdk-js-v3/pull/4427/commits/3e1cb31c48c5b75a90cbd69f9e638070afecaaa1).